src/rules/enforce-boolean-naming-prefixes.ts

```
@@ -399,6 +394,47 @@ export const enforceBooleanNamingPrefixes = createRule<Options, MessageIds>({
 
+    /**
+     * Check if a function call returns a boolean value based on its name
+     */
+    function isBooleanReturningFunction(calleeName: string): boolean {
+      // Special handling for "assert" functions - distinguish between validation and boolean assertion functions
+      if (calleeName.toLowerCase().startsWith('assert')) {
+        // Validation/assertion functions that return the validated input (not boolean)
+        // These typically have names like: assertSafe, assertValid, assertString, assertNumber, etc.
+        const validationAssertPatterns = [
+          'assert'
+        ];
```

ESLint

[error] Replace `⏎··········'assert'⏎········` with `'assert'`

(prettier/prettier)